modules/cudf/src/column/reduction.cpp

```diff
@@ -46,86 +47,98 @@ cudf::data_type _compute_dtype(cudf::type_id id) {
+  } catch (std::exception const& e) { NAPI_THROW(Napi::Error::New(Env(), e.what())); }
 }
 
-ObjectUnwrap<Scalar> Column::sum(rmm::mr::device_memory_resource* mr) const {
+Scalar::wrapper_t Column::sum(rmm::mr::device_memory_resource* mr) const {
   return reduce(cudf::make_sum_aggregation(), _compute_dtype(this->type().id()), mr);
 }
 
 Napi::Value Column::sum(Napi::CallbackInfo const& info) {
-  return sum(NapiToCPP(info[0]).operator rmm::mr::device_memory_resource*());
+  return Napi::Value::From(info.Env(),
+                           sum(NapiToCPP(info[0]).operator rmm::mr::device_memory_resource*()));
 }
 
-ObjectUnwrap<Scalar> Column::product(rmm::mr::device_memory_resource* mr) const {
+Scalar::wrapper_t Column::product(rmm::mr::device_memory_resource* mr) const {
   return reduce(cudf::make_product_aggregation(), _compute_dtype(this->type().id()), mr);
 }
 
 Napi::Value Column::product(Napi::CallbackInfo const& info) {
-  return product(NapiToCPP(info[0]).operator rmm::mr::device_memory_resource*());
+  return Napi::Value::From(info.Env(),
+                           product(NapiToCPP(info[0]).operator rmm::mr::device_memory_resource*()));
 }
 
-ObjectUnwrap<Scalar> Column::sum_of_squares(rmm::mr::device_memory_resource* mr) const {
+Scalar::wrapper_t Column::sum_of_squares(rmm::mr::device_memory_resource* mr) const {
   return reduce(cudf::make_sum_of_squares_aggregation(), _compute_dtype(this->type().id()), mr);
 }
 
 Napi::Value Column::sum_of_squares(Napi::CallbackInfo const& info) {
-  return sum_of_squares(NapiToCPP(info[0]).operator rmm::mr::device_memory_resource*());
+  return Napi::Value::From(
+    info.Env(), sum_of_squares(NapiToCPP(info[0]).operator rmm::mr::device_memory_resource*()));
 }
 
 Napi::Value Column::any(Napi::CallbackInfo const& info) {
   CallbackArgs args{info};
-  return reduce(cudf::make_any_aggregation(), cudf::data_type(cudf::type_id::BOOL8), args[0]);
+  return Napi::Value::From(
+    info.Env(),
+    reduce(cudf::make_any_aggregation(), cudf::data_type(cudf::type_id::BOOL8), args[0]));
 }
 
 Napi::Value Column::all(Napi::CallbackInfo const& info) {
   CallbackArgs args{info};
-  return reduce(cudf::make_all_aggregation(), cudf::data_type(cudf::type_id::BOOL8), args[0]);
+  return Napi::Value::From(
+    info.Env(),
+    reduce(cudf::make_all_aggregation(), cudf::data_type(cudf::type_id::BOOL8), args[0]));
 }
 
-ObjectUnwrap<Scalar> Column::mean(rmm::mr::device_memory_resource* mr) const {
+Scalar::wrapper_t Column::mean(rmm::mr::device_memory_resource* mr) const {
   return reduce(cudf::make_mean_aggregation(), cudf::data_type(cudf::type_id::FLOAT64), mr);
 }
 
 Napi::Value Column::mean(Napi::CallbackInfo const& info) {
-  return mean(NapiToCPP(info[0]).operator rmm::mr::device_memory_resource*());
+  return Napi::Value::From(info.Env(),
+                           mean(NapiToCPP(info[0]).operator rmm::mr::device_memory_resource*()));
```

Comment:
I noticed some of the `_re` functions above switched to using `CallbackArgs args{info}` and avoided the explicit 
```
NapiToCPP(info[0]).operator rmm::mr::device_memory_resource
```
As magic goes, that's one bit I could live with, is there a reason not to use `CallbackArgs args{info}` more liberally (or even conventionally)?

Reply:
Yeah, I can see an argument to be made for consistency here. I usually only construct a `CallbackArgs` if I need to convert multiple arguments, i.e.:

```c++
CallbackArgs const args{info}
foo(args[0], args[1], args[2]);
```

But I've also used it as a shortcut for a single argument before:

```c++
bar(CallbackArgs{info}[0].operator rmm::mr::device_memory_resource*());
// vs.
bar(NapiToCPP{info[0]}.operator rmm::mr::device_memory_resource*());
```

I like the idea of only using `CallbackArgs`, and making `NapiToCPP` an internal implementation detail of `CallbackArgs`. I'll also do some tests to see if I can make `CallbackArgs` a drop-in replacement for `Napi::CallbackInfo` so we can use it in methods defined via the ObjectWrap APIs.